### ISSUE TITLE
Loading files as parameter fails using multiple config files with '<()'

### ISSUE THREAD

**Issue by Fabbbrrr** [cloudtools/stacker#482]
We are trying to use the file lookup to load some external files for cfn-init/userdata and found a problem with our setup. A few weeks ago, a colleague raised this question regarding multiple configs: https://github.com/remind101/stacker/issues/403

The solution was just to use '<()' in the bash script to load multiple config files and pass a file descriptor to stacker.

It worked well until we started using file lookups in the YAML file as 

`MyVariable: ${file parameterized-b64:file://path/to/file}`

The problem seems to be that unless I specify the absolute path to my file, from the root of my filesystem, it won't pick up the file, throwing an error like this:

> Traceback (most recent call last):
>   File "/usr/local/bin/stacker", line 9, in <module>
>     args.run(args)
>   File "/usr/local/lib/python2.7/site-packages/stacker/commands/stacker/build.py", line 52, in run
>     dump=options.dump)
>   File "/usr/local/lib/python2.7/site-packages/stacker/actions/base.py", line 127, in execute
>     self.run(*args, **kwargs)
>   File "/usr/local/lib/python2.7/site-packages/stacker/actions/build.py", line 310, in run
>     provider=self.provider)
>   File "/usr/local/lib/python2.7/site-packages/stacker/plan.py", line 323, in dump
>     provider=provider,
>   File "/usr/local/lib/python2.7/site-packages/stacker/stack.py", line 153, in resolve
>     resolve_variables(self.variables, context, provider)
>   File "/usr/local/lib/python2.7/site-packages/stacker/variables.py", line 77, in resolve_variables
>     variable.resolve(context, provider)
>   File "/usr/local/lib/python2.7/site-packages/stacker/variables.py", line 143, in resolve
>     raise FailedVariableLookup(self.name, e)
> stacker.exceptions.FailedVariableLookup: Couldn't resolve lookups in variable `UserData`. [Errno 2] No such file or directory: '/dev/fd/path/to/file'
> 

If you look closely, /dev/fd/ is appended to whatever file I specify in the lookup configuration. We have debugged the stacker code and found that in stacker/util.py get_config_directory() stacker is not returning the actual config directory, instead it returns the file descriptor.

To sum up, the recommended way of loading multiple config files is not considering the way the file lookups are working. This is the first time it happens to us but I believe it will be a very common scenario from now on. 

I'd appreciate your input on this one.

**Comment by phobologic** [comment]
Hey @Fabbbrrr - yeah, that issue makes a lot of sense.  Honestly, there's probably not a good answer to this out of the box.  A more genericized version of this would be pretty difficult actually - if you have multiple config files, and they are in different directories, how do I choose the root directory to look for the `file` lookup?  We could say only the first config.

All that said, I think something is in the works that should fit what you were asking for in #403.  #458 provides remote/layered configs.  Probably worth checking it out to see if it fits your use case.

**Comment by ejholmes** [comment]
Should the `${file ...}` lookup just use `$PWD` as the base path?

@Fabbbrrr if you're already generating a stacker config in a `<()`, you could just inline the contents of the file there, instead of using stacker's `${file }` lookup.

**Comment by phobologic** [comment]
Haven't heard back, going to go ahead and close this issue.  Please re-open if still interested!
